arch/posix/CMakeLists.txt

```diff
@@ -1,12 +1,59 @@
 # SPDX-License-Identifier: Apache-2.0
 
-if (CONFIG_64BIT)
-	# some gcc versions fail to build without -fPIC
-	zephyr_compile_options(-m64 -fPIC)
-	zephyr_link_libraries(-m64)
-else ()
-	zephyr_compile_options(-m32)
-	zephyr_link_libraries(-m32)
+# @Intent: Detect host architecture (see 'man 1 arch')
+execute_process(
+  COMMAND
+  arch
+  OUTPUT_VARIABLE
+  BUILD_ARCH
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+
+# @Intent: Set any *necessary* arch-specific compile / linker options
+if (${BUILD_ARCH} STREQUAL "aarch64")
+
+  # For Aarch64, multilib is not an actively pursued solution for with most
+  # distributions. Userspace is (generally) either 32-bit or 64-bit but not
+  # both.
+
+  # @Intent: Call a script to get userspace wordsize for comparison with CONFIG_64BIT
+  execute_process(
+    COMMAND
+    ${PYTHON_EXECUTABLE}
+    ${ZEPHYR_BASE}/scripts/user_wordsize.py
+    OUTPUT_VARIABLE
+    AARCH64_USERSPACE_BITS
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
+  if (CONFIG_64BIT)
+    if (${AARCH64_USERSPACE_BITS} STREQUAL "32")
+      message(WARNING
+        "CONFIG_64BIT=y but this Aarch64 machine has a 32-bit userspace.\n"
+        "Setting COINFIG_64BIT=n"
+      )
+      SET(CONFIG_64BIT n CACHE BOOL "CONFIG_64BIT=n" FORCE)
```

Comment:
error out also means there is no need to pollute the CMake cache with this variable, which is actually a Kconfig setting.